### PR TITLE
Added Even Odd in Java

### LINKED ISSUE #936: Add Even Odd in Java

### PR DESCRIPTION
**Pull request by auroq**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language][1] repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [ ] I plan to write the article for this code snippet
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [x] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])
- [x] I fixed #936 

## Notes

Feel free to put whatever you want here.

[1]: https://therenegadecoder.com/code/sample-programs-in-every-language/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#plagiarism

### CODE REVIEW

**Comment by auroq** [comment]
I used your suggestion for the C# one (#890) in this implementation. It only parses the last character to an int. That way we can use any length of number. I'll have to change the C# implementation to do this, too. Maybe we should even make that a requirement, and change all implementations that way.

**Comment by jrg94** [comment]
Also, I'd like to echo your suggestion of updating the requirements. Once we settle on the requirements, we can go back and fix things up.

**Comment by auroq** [comment]
I agree. I think we do need to verify that the input is a number. We got that for free when we were parsing the hole string to a number, but we lost it by looking at just the last digit.

Take a look at this verification function and let me know what you think.

File: archive/j/java/EvenOdd.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+public class EvenOdd
+{
+    public static void verifyNumber(String n) throws Exception
+    {
+        char[] nArray = n.toCharArray();
+        for (char c : nArray)
+        {
+            if (!Character.isDigit(c))
+                throw new Exception();
```
Comment (jrg94) [inline archive/j/java/EvenOdd.java]: Can you try throwing something more specific like https://docs.oracle.com/javase/8/docs/api/java/lang/NumberFormatException.html?

File: archive/j/java/EvenOdd.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        try
+        {
+            String nstr = args[0].trim();
+            verifyNumber(nstr);
+            String lastDigit = nstr.substring(nstr.length() - 1);
+            int n = Integer.parseInt(lastDigit);
+            String result = n % 2 == 0 ? "Even" : "Odd";
+            System.out.println(result);
+        }
+        catch (Exception e)
```
Comment (jrg94) [inline archive/j/java/EvenOdd.java]: Reflect the change from above here. I believe parseInt throws the same error.

**Comment by auroq** [comment]
As much as I dislike the way Java makes you do separate catch statements for each error type, you're probably right that more specific exceptions is better. We have to catch the number format for the reasons you mentioned above. We also have to catch the index out of bounds for the case where no input is passed. I also added handling for negative numbers.

**Comment by jrg94** [comment]
I think you can simplify the catches like:

`(NumberFormatException | IndexArrayOutOfBoundsException e)`

Since they do the same thing.

**Comment by jrg94** [comment]
No sweat! I'm a bit more tough on Java submissions since I teach it. 😄

**Comment by auroq** [comment]
Think I got it this time.

**Pull request merged**